charts/ark-pentaho-ce/files/context.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--===========================================================================
+HITACHI VANTARA PROPRIETARY AND CONFIDENTIAL
+
+Copyright 2002 - 2017 Hitachi Vantara. All rights reserved.
+
+NOTICE: All information including source code contained herein is, and
+remains the sole property of Hitachi Vantara and its licensors. The intellectual
+and technical concepts contained herein are proprietary and confidential
+to, and are trade secrets of Hitachi Vantara and may be covered by U.S. and foreign
+patents, or patents in process, and are protected by trade secret and
+copyright laws. The receipt or possession of this source code and/or related
+information does not convey or imply any rights to reproduce, disclose or
+distribute its contents, or to manufacture, use, or sell anything that it
+may describe, in whole or in part. Any reproduction, modification, distribution,
+or public display of this information without the express written authorization
+from Hitachi Vantara is strictly prohibited and in violation of applicable laws and
+international treaties. Access to the source code contained herein is strictly
+prohibited to anyone except those individuals and entities who have executed
+confidentiality and non-disclosure agreements or other agreements with Hitachi Vantara,
+explicitly covering such access.
+============================================================================-->
+<Context>
+    <Resource
+        name="jdbc/acm3DataSource"
+        auth="Container"
+        type="javax.sql.DataSource"
+        factory="org.apache.commons.dbcp.BasicDataSourceFactory"
+        maxTotal="20"
+        maxIdle="5"
+        maxWaitMillis="10000"
+        username="${DB_USERNAME}"

```

Comment:
Could you please explain to me how environment variables get expanded here? Or more exactly: which software of piece of software does this expansion? Thanks

Reply:
All right, thanks a lot @naveen-armedia, I understand much better now. In the first link, it says that expansion of environment variables is supported since 7.0, so I suppose we should be good?

In the links you provided, I saw this a couple of times:

> Environment variables can be referenced in the server.xml etc by setting the system property org.apache.tomcat.util.digester.PROPERTY_SOURCE to the value org.apache.tomcat.util.digester.Digester$EnvironmentPropertySource

But it looks like this is not done for pentaho-ce... Would you have any comment on this?

Reply:
Actually we tried to keep this property in catalina.properties file but unfortunately this is not provided in 8.5.27 please check [https://github.com/apache/tomcat/tree/8.5.27/java/org/apache/tomcat/util/digester](https://github.com/apache/tomcat/tree/8.5.27/java/org/apache/tomcat/util/digester) here there is no class file `EnvironmentPropertySource`

Reply:
Oh, OK... What are you going to do, then?

Reply:
At this moment we will go with what we have. And also as we have planned to use parameterization most of the config files will go with the docker image. But I agree with the cause of concern but we will get to it soon. Thanks!!

Reply:
I'm not sure I understand... Are you saying that the environment variable substitution is working? I thought the whole thread was about the environment variable substitution not working?...

Reply:
@naveen-armedia If you could explain to me what is happening here, that would be great. I will then mark the conversation as solved. Thanks.

Reply:
@fabricetriboix  tomcat 8.5.27 version does not replace environment variables in most of the configuration files but here it is working for context.xml configuration.

Reply:
@naveen-armedia All right... That is most bizarre... Anyway, I will mark the conversation as closed.